mock.go

```diff
@@ -274,12 +275,28 @@ func (t *MockTable) zero() interface{} {
 	return reflect.New(reflect.TypeOf(t.entity)).Interface()
 }
 
-func (t *MockTable) keyFromColumnValues(values map[string]interface{}, keyNames []string) (key, error) {
+func (t *MockTable) partitionKeyFromColumnValues(values map[string]interface{}, keyNames []string) (key, error) {
 	var key key
 
 	for _, keyName := range keyNames {
 		value, ok := values[keyName]
+		// need to explcitly check for string type here, as we cannot
+		// write an empty string in the primary key
+		stringVal, isString := value.(string)
+		if !ok || (isString && stringVal == "") {
+			return nil, fmt.Errorf("Missing mandatory PRIMARY KEY part %s", keyName)
+		}
+		key = key.Append(keyName, value)
+	}
+
+	return key, nil
+}
+
+func (t *MockTable) clusteringKeyFromColumnValues(values map[string]interface{}, keyNames []string) (key, error) {
```

Comment:
I've separated out validation for `clusteringKeys` and `partitionKeys`, to allow clustering keys to be empty.